### ISSUE TITLE
Syntax Sugar

### ISSUE THREAD

**Issue by facekapow** [alta-lang/alta-core#11]
As part of the QoL wave designed to make life easier for users, there's a couple of things that could be added with regards to syntax sugar:

## Classic `for` loops
These are your regular, everyday, all-purpose `for` loops, and they're absolutely essential.
```alta
for (let i: int = 0; i < 10; i++) {
  # body
}
```
## Ranged `for` loops
These are simple loops designed for iteration, with the bonus that they're easy to implement in the back-end, useful, and pleasant for users (particularly because they give indexes while looking and feeling like Java's "enhanced for loops").
```alta
# 0 included; 10 excluded; incrementing
for i: int in 0..10 {
  # body
}
# 0 included; 10 included; incrementing
for i: int in 0...10 {
  # body
}

# 10 included; 0 excluded; decrementing
for i: int in 10::0 {
  # body
}
# 10 included; 0 included; decrementing
for i: int in 10:::0 {
  # body
}
```

## `when` block
The idea is that they'll be more powerful than classic `switch` blocks, but will not have some of the harder-to-grasp features of `switch`es (like cascading scopes/executions). They'll support various different kinds of verbal comparisons, but for the cases where it doesn't provide a verbal alternative, they'll have `makes` conditions to evaluate expressions as booleans. They'll also have their input assigned to the `$` variable, for easy reference. The first condition that matches is the one they execute.

Basically, they're designed to replace the common (but repetitive) `if ... else if ... else if ...` pattern when checking a single value against multiple conditions.
```alta
when foobar {
  is 4 {
    # do something
  }
  isn't 6 {
    # something else
  }
  is > 5 {
   # blah blah
  }
  makes foo(8, $, bar) == true {
    # ooh, look
  }
}
```

**Comment by facekapow** [comment]
Thought of some more:

## Post-fix and pre-fix increment and decrement
Something every language should have.
```alta
# prefix increment
++i

# postfix increment
i++

# prefix decrement
--i

# postfix decrement
i--
```

**Comment by facekapow** [comment]
The v0.11.0 release is now feature-complete (only last-minute bug fixes will be added), and most of these are done, with the exception of the `when` block, which is now planned for v0.12.0.
